addons/sourcemod/scripting/shop.sp

```diff
@@ -1198,6 +1202,238 @@ public int ItemPanel_Handler(Menu menu, MenuAction action, int param1, int param
 	}
 }
 
+void ConfirmBuy(int client, int item_id)
+{
+	Panel panel = ItemManager_ConfirmItemPanelInfo(client, item_id, Menu_Buy, true);
+	if (panel != null)
+	{
+		char sBuffer[SHOP_MAX_STRING_LENGTH], sItemId[16];
+		IntToString(item_id, sItemId, sizeof(sItemId));
+		
+		bool isHidden = ItemManager_GetItemHideEx(sItemId);
+		
+		SetGlobalTransTarget(client);
+		
+		int credits = GetCredits(client);
+		
+		FormatEx(sBuffer, sizeof(sBuffer), "%t\n ", "credits", credits);
+		panel.SetTitle(sBuffer, false);
+		
+		ItemType type = ItemManager_GetItemTypeEx(sItemId);
+	
+		panel.DrawItem(" ", ITEMDRAW_SPACER|ITEMDRAW_RAWLINE);
+		
+		switch (type)
+		{
+			case Item_None :
+			{
+				if (!isHidden)
+				{
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "You Buy Sure");
+					panel.DrawText(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "Yes");
+					panel.DrawItem(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "No");
+					panel.DrawItem(sBuffer);
+				}
+			}
+			case Item_Finite :
+			{
+				int count = PlayerManager_GetItemCountEx(client, sItemId);
+				FormatEx(sBuffer, sizeof(sBuffer), "%t: %d", "You have", count);
+				panel.DrawText(sBuffer);
+				
+				panel.DrawItem(" ", ITEMDRAW_SPACER|ITEMDRAW_RAWLINE);
+				
+				if (!isHidden)
+				{
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "You Buy Sure");
+					panel.DrawText(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "Yes");
+					panel.DrawItem(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "No");
+					panel.DrawItem(sBuffer);
+				}
+			}
+			case Item_Togglable :
+			{
+				if (!isHidden)
+				{
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "You Buy Sure");
+					panel.DrawText(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "Yes");
+					panel.DrawItem(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "No");
+					panel.DrawItem(sBuffer);
+				}
+			}
+			case Item_BuyOnly :
+			{
+				if (!isHidden)
+				{
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "You Buy Sure");
+					panel.DrawText(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "Yes");
+					panel.DrawItem(sBuffer);
+
+					FormatEx(sBuffer, sizeof(sBuffer), "%t", "No");
+					panel.DrawItem(sBuffer);
+				}
+			}
+		}
```

Comment:
So if item is `Togglable` or `BuyOnly` and is hidden, you're sending panel without any buttons, only with title. Strange behavior.